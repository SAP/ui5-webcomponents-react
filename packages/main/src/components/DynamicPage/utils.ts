export const DynamicPageCssVariables = {
  headerDisplay: '--_ui5wcr_DynamicPage_header_display',
  titleFontSize: '--_ui5wcr_DynamicPage_title_fontsize'
};
