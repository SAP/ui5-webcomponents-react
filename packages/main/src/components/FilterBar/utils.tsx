import { cloneElement } from 'react';

const inputTagNames = new Set([
  'UI5-COMBOBOX',
  'UI5-DATE-PICKER',
  'UI5-DATERANGE-PICKER',
  'UI5-DATETIME-PICKER',
  'UI5-DURATION-PICKER',
  'UI5-INPUT',
  'UI5-MULTI-INPUT',
  'UI5-RATING-INDICATOR',
  'UI5-STEP-INPUT',
  'UI5-TIME-PICKER'
]);

export const filterValue = (ref, child) => {
  const tagName = ref.tagName;
  let filterItemProps = {};
  if (inputTagNames.has(tagName)) {
    filterItemProps = { value: ref.value };
  }
  if (tagName === 'UI5-SELECT' || tagName === 'UI5-MULTI-COMBOBOX') {
    const selectedIndices = Array.from(ref.children as HTMLCollectionOf<any>)
      .map((item, index) => (item.selected ? index : false))
      .filter((el) => el !== false);
    const selectedIndicesSet = new Set(selectedIndices);
    const options = child.props.children.props.children?.map((item, index) => {
      if (selectedIndicesSet.has(index)) {
        return cloneElement(item, { selected: true });
      }
      return cloneElement(item, { selected: false });
    });
    filterItemProps = { children: options };
  }
  if (tagName === 'UI5-SWITCH' || tagName === 'UI5-CHECKBOX') {
    filterItemProps = { checked: ref.checked };
  }
  return filterItemProps;
};

export const renderSearchWithValue = (renderSearchElement, searchValue) => {
  return cloneElement(renderSearchElement, { value: searchValue ?? renderSearchElement?.props?.value });
};
