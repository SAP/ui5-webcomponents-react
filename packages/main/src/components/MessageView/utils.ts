import iconAlert from '@ui5/webcomponents-icons/dist/alert.js';
import iconError from '@ui5/webcomponents-icons/dist/error.js';
import iconInformation from '@ui5/webcomponents-icons/dist/information.js';
import iconSysEnter from '@ui5/webcomponents-icons/dist/sys-enter-2.js';
import { ValueState } from '../../enums/ValueState';

export const getIconNameForType = (type: ValueState | keyof typeof ValueState): string => {
  switch (type) {
    case ValueState.Error:
    case 'Error':
      return iconError;
    case ValueState.Success:
    case 'Success':
      return iconSysEnter;
    case ValueState.Warning:
    case 'Warning':
      return iconAlert;
    default:
      return iconInformation;
  }
};
