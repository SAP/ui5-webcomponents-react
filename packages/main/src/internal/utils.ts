import { Children, Fragment, ReactNode } from 'react';

export function flattenFragments(children: ReactNode | ReactNode[], depth = 1): ReactNode[] {
  const flatChildren = [];
  if (Array.isArray(children)) {
    return children;
  }
  const removeFragments = (element, level = 0) => {
    if (!element) {
      return;
    }
    if (element.type === Fragment && level <= depth) {
      Children.toArray(element.props?.children).forEach((item) => {
        removeFragments(item, level + 1);
      });
    } else {
      flatChildren.push(element);
    }
  };
  removeFragments(children);

  return flatChildren;
}
