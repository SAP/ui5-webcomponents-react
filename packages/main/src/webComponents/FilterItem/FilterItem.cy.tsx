import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { FilterItem } from './index';

describe('FilterItem', () => {
  it('Basic Test (generated)', () => {
    cy.mount(<FilterItem />);
    cy.get('[ui5-filter-item]').should('exist');
  });

  cypressPassThroughTestsFactory(FilterItem);
});
