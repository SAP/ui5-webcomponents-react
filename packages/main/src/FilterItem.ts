import { deprecationNotice } from '@ui5/webcomponents-react-base';
import type { FilterItemDomRef, FilterItemPropTypes } from './webComponents/FilterItem';
import { FilterItem } from './webComponents/FilterItem';

deprecationNotice(
  'FilterItem',
  `importing from '@ui5/webcomponents-react/dist/FilterItem' is deprecated. Please use "import { FilterItem } from '@ui5/webcomponents-react';" instead.`
);

export { FilterItem };

export type { FilterItemPropTypes, FilterItemDomRef };
