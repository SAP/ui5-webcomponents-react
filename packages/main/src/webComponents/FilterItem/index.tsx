import '@ui5/webcomponents-fiori/dist/FilterItem.js';
import { ReactNode } from 'react';
import { CommonProps } from '../../interfaces/CommonProps';
import { Ui5DomRef } from '../../interfaces/Ui5DomRef';
import { withWebComponent } from '../../internal/withWebComponent';

interface FilterItemAttributes {
  /**
   * Defines the text of the component.
   */
  text?: string;
}

export interface FilterItemDomRef extends FilterItemAttributes, Ui5DomRef {}

export interface FilterItemPropTypes extends FilterItemAttributes, CommonProps {
  /**
   * Defines the `values` list.
   *
   * __Note:__ When passing a custom React component to this prop, you have to make sure your component reads the `slot` prop and appends it to the most outer element of your component.
   * Learn more about it [here](https://sap.github.io/ui5-webcomponents-react/?path=/docs/knowledge-base--page#adding-custom-components-to-slots).
   */
  values?: ReactNode | ReactNode[];
}

/**
 *
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/FilterItem" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const FilterItem = withWebComponent<FilterItemPropTypes, FilterItemDomRef>(
  'ui5-filter-item',
  ['text'],
  [],
  ['values'],
  []
);

FilterItem.displayName = 'FilterItem';

export { FilterItem };
