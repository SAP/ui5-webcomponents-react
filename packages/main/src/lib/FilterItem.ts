import { FilterItem } from '../webComponents/FilterItem';
import type { FilterItemPropTypes } from '../webComponents/FilterItem';

export { FilterItem };
export type { FilterItemPropTypes };
