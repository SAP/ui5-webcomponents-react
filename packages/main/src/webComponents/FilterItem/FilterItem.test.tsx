import { render } from '@shared/tests';
import { createCustomPropsTest } from '@shared/tests/utils';
import { FilterItem } from '@ui5/webcomponents-react/dist/FilterItem';
import React from 'react';

describe('FilterItem', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(<FilterItem />);
    expect(asFragment()).toMatchSnapshot();
  });
  createCustomPropsTest(FilterItem);
});
