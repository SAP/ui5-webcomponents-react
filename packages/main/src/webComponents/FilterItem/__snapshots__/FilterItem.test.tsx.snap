// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItem Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-filter-item
    ui5-filter-item=""
  />
</DocumentFragment>
`;
